src/documentation/programming-guide.md

```diff
@@ -1095,24 +1531,41 @@ The following code sample shows how to override `expand` for the `ComputeWordLen
 {% github_sample /apache/beam/blob/master/sdks/python/apache_beam/examples/snippets/snippets_test.py tag:model_composite_transform
 %}```
 
-As long as you override the `expand` method in your `PTransform` subclass to accept the appropriate input `PCollection`(s) and return the corresponding output `PCollection`(s), you can include as many transforms as you want. These transforms can include core transforms, composite transforms, or the transforms included in the Beam SDK libraries.
-
-**Note:** The `expand` method of a `PTransform` is not meant to be invoked directly by the user of a transform. Instead, you should call the `apply` method on the `PCollection` itself, with the transform as an argument. This allows transforms to be nested within the structure of your pipeline.
-
-#### PTransform Style Guide
+As long as you override the `expand` method in your `PTransform` subclass to
+accept the appropriate input `PCollection`(s) and return the corresponding
+output `PCollection`(s), you can include as many transforms as you want. These
+transforms can include core transforms, composite transforms, or the transforms
+included in the Beam SDK libraries.
 
-When you create a new `PTransform`, be sure to read the [PTransform Style Guide]({{ site.baseurl }}/contribute/ptransform-style-guide/). The guide contains additional helpful information such as style guidelines, logging and testing guidance, and language-specific considerations.
+**Note:** The `expand` method of a `PTransform` is not meant to be invoked
+directly by the user of a transform. Instead, you should call the `apply` method
+on the `PCollection` itself, with the transform as an argument. This allows
+transforms to be nested within the structure of your pipeline.
 
-## <a name="io"></a>Pipeline I/O
+#### 4.6.3. PTransform Style Guide
 
-When you create a pipeline, you often need to read data from some external source, such as a file in external data sink or a database. Likewise, you may want your pipeline to output its result data to a similar external data sink. Beam provides read and write transforms for a [number of common data storage types]({{site.baseurl }}/documentation/io/built-in/). If you want your pipeline to read from or write to a data storage format that isn't supported by the built-in transforms, you can [implement your own read and write transforms]({{site.baseurl }}/documentation/io/io-toc/).
+When you create a new `PTransform`, be sure to read the [PTransform Style
+Guide]({{ site.baseurl }}/contribute/ptransform-style-guide/). The guide
```

Comment:
So demanding!

Maybe "The PTransform style guide contains additional information not included here such as...", "It is a useful starting point when writing new composite PTransforms"?